### PR TITLE
Added tests for DefaultZoneConfiguration

### PR DESCRIPTION
**Pull request by sanjeevchopra**

### CODE REVIEW

File: src/main/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguration.java [hunk 1]
```diff
@@ -25,17 +25,16 @@
...
 
 public class DefaultZoneConfiguration implements InitializingBean {
-    private String trustedIssuerId;
     private List<String> allowedUriPatterns;
     private List<String> trustedIssuerIds = new ArrayList<>();
 
+    // This field is for backward compatibility.
     public String getTrustedIssuerId() {
-        return this.trustedIssuerId;
+        return this.trustedIssuerIds.isEmpty() ? null : this.trustedIssuerIds.get(0);
```
Comment (irinaepshteyn) [inline src/main/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguration.java]: Add tests for setting trusted issuer(s) with null and empty list and getting trusted issuer(s) when not configured.

Comment (sanjeevchopra) [reply src/main/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguration.java]: thanks Irina, i added the tests. to be fair those should've been there to begin with :)

File: src/main/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguration.java [hunk 2]
```diff
@@ -25,17 +25,16 @@
...
     }
 
     public void setTrustedIssuerId(final String trustedIssuerId) {
-        this.trustedIssuerId = trustedIssuerId;
-        // providing default behavior for existing single issuer contract
+        this.trustedIssuerIds = new ArrayList<>(); //replace list
```
Comment (irinaepshteyn) [inline src/main/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguration.java]: Is it possible to add null to the ArrayList?  Will adding null increase the size to 1?  If so, we need to check for null here and not add.  Also add a test.

File: src/test/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguratorTest.java [hunk 3]
```diff
@@ -61,15 +62,38 @@ public void testMultipleTrustedIssuers() {
...
+        DefaultZoneConfiguration configurator = new DefaultZoneConfiguration();
+        configurator.setTrustedIssuerIds(null);
+
+        Assert.assertEquals(configurator.getTrustedIssuerIds(), Collections.emptyList());
+        Assert.assertEquals(configurator.getTrustedIssuerId(), null);
+    }
+
+    public void testSetTrustedIssuerIdsEmpty() {
+        DefaultZoneConfiguration configurator = new DefaultZoneConfiguration();
+        configurator.setTrustedIssuerIds(new ArrayList<String>());
```
Comment (irinaepshteyn) [inline src/test/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguratorTest.java]: Could use Collections.emptyList() instead "new ArrayList<String>()".

File: src/main/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguration.java [hunk 4]
```diff
@@ -25,18 +25,19 @@
...
     }
 
     public void setTrustedIssuerId(final String trustedIssuerId) {
-        this.trustedIssuerId = trustedIssuerId;
-        // providing default behavior for existing single issuer contract
-        this.trustedIssuerIds.add(trustedIssuerId);
+        if (null != trustedIssuerId) {
+            this.trustedIssuerIds = new ArrayList<>(); // replace list
+            this.trustedIssuerIds.add(trustedIssuerId);
+        }
```
Comment (irinaepshteyn) [inline src/main/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguration.java]: To be consistent with setTrustedIssuerIds, replace existing list if null is passed.

Comment (sanjeevchopra) [reply src/main/java/com/ge/predix/uaa/token/lib/DefaultZoneConfiguration.java]: done
